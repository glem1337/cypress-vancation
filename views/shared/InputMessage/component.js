import PropTypes from 'prop-types';
import classNames from 'classnames';
import FormattedOrRawMessage from 'views/shared/FormattedOrRawMessage';

const InputMessage = ({
  icon, message,
}) => {
  const iconClassNames = classNames(
    'main-input__message-icon icon',
    {
      'icon-close': icon === 'close',
      'icon-alert': icon === 'alert',
      'icon-info': icon === 'info',
      'icon-tick': icon === 'tick',
    },
  );

  return (
    <p className="main-input__message">
      <i className={iconClassNames} />
      <FormattedOrRawMessage message={message} />
    </p>
  );
};

InputMessage.defaultProps = {
  icon: null,
};

InputMessage.propTypes = {
  icon: PropTypes.oneOf(['close', 'alert', 'info', 'tick', null]),
  message: PropTypes.oneOfType([
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      values: PropTypes.shape(),
    }),
    PropTypes.string,
  ]).isRequired,
};

export default InputMessage;
