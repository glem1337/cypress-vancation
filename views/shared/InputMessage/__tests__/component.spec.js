import { shallow } from 'enzyme';

import InputMessage from '../component';

describe('InputMessage component matches snapshot', () => {
  const defaultProps = {
    message: { id: 'test.message' },
  };

  it('with default props', () => {
    const wrapper = shallow(<InputMessage {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('with `icon: close`', () => {
    const props = {
      ...defaultProps,
      icon: 'close',
    };
    const wrapper = shallow(<InputMessage {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('with `icon: alert`', () => {
    const props = {
      ...defaultProps,
      icon: 'alert',
    };
    const wrapper = shallow(<InputMessage {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('with `icon: info`', () => {
    const props = {
      ...defaultProps,
      icon: 'info',
    };
    const wrapper = shallow(<InputMessage {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('with `icon: tick`', () => {
    const props = {
      ...defaultProps,
      icon: 'tick',
    };
    const wrapper = shallow(<InputMessage {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('with `icon: null`', () => {
    const props = {
      ...defaultProps,
      icon: null,
    };
    const wrapper = shallow(<InputMessage {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
