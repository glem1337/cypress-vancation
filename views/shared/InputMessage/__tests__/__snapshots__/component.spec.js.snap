// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputMessage component matches snapshot with \`icon: alert\` 1`] = `
<p
  className="main-input__message"
>
  <i
    className="main-input__message-icon icon icon-alert"
  />
  <FormattedOrRawMessage
    message={
      Object {
        "id": "test.message",
      }
    }
  />
</p>
`;

exports[`InputMessage component matches snapshot with \`icon: close\` 1`] = `
<p
  className="main-input__message"
>
  <i
    className="main-input__message-icon icon icon-close"
  />
  <FormattedOrRawMessage
    message={
      Object {
        "id": "test.message",
      }
    }
  />
</p>
`;

exports[`InputMessage component matches snapshot with \`icon: info\` 1`] = `
<p
  className="main-input__message"
>
  <i
    className="main-input__message-icon icon icon-info"
  />
  <FormattedOrRawMessage
    message={
      Object {
        "id": "test.message",
      }
    }
  />
</p>
`;

exports[`InputMessage component matches snapshot with \`icon: null\` 1`] = `
<p
  className="main-input__message"
>
  <i
    className="main-input__message-icon icon"
  />
  <FormattedOrRawMessage
    message={
      Object {
        "id": "test.message",
      }
    }
  />
</p>
`;

exports[`InputMessage component matches snapshot with \`icon: tick\` 1`] = `
<p
  className="main-input__message"
>
  <i
    className="main-input__message-icon icon icon-tick"
  />
  <FormattedOrRawMessage
    message={
      Object {
        "id": "test.message",
      }
    }
  />
</p>
`;

exports[`InputMessage component matches snapshot with default props 1`] = `
<p
  className="main-input__message"
>
  <i
    className="main-input__message-icon icon"
  />
  <FormattedOrRawMessage
    message={
      Object {
        "id": "test.message",
      }
    }
  />
</p>
`;
